subsys/net/lib/mqtt_sn/mqtt_sn_transport_udp.c

```diff
@@ -43,25 +44,65 @@ static int tp_udp_init(struct mqtt_sn_transport *transport)
 {
 	struct mqtt_sn_transport_udp *udp = UDP_TRANSPORT(transport);
 	int err;
+	struct sockaddr_in addrm;
+	struct ip_mreqn mreqn;
+	int optval;
+	struct net_if *iface;
 
-	udp->sock = zsock_socket(udp->gwaddr.sa_family, SOCK_DGRAM, 0);
+	udp->sock = zsock_socket(udp->bcaddr.sa_family, SOCK_DGRAM, 0);
 	if (udp->sock < 0) {
 		return errno;
 	}
 
 	LOG_DBG("Socket %d", udp->sock);
 
+	/* Enable broadcasting mode */
+	/*
+	 * int broadcastEnable = 1;
```

Comment:
Please remove commented out code.

Reply:
I was hoping to get the opinion of someone who knows Linux's networking stack better whether this is required or not? I was able to test it and it worked without this but unsure if that is correct or not.

Reply:
Well, we don't even support `SO_BROADCAST` socket option in Zephyr so I guess the answer is it's not needed (at least with Zephyr stack).